Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -4846,6 +4848,138 @@ def panorama_override_vulnerability(threatid: str, vulnerability_profile: str, d
+        'key': API_KEY,
+        'cmd': "<show><interface>all</interface></show>"
+    }
+
+    return http_request(
+        URL,
+        'GET',
+        params=params)
+
+
+def panorama_route_lookup(dest_ip: str, virtual_router=None):
+    """
+    Runs a test-routing fib-lookup to determine the outgoing interface of a given IP address.
+    """
+    params = {
+        'type': 'op',
+        'key': API_KEY,
+        'cmd': f"<test><routing><fib-lookup><ip>{dest_ip}</ip><virtual-router>{virtual_router}</virtual-router>"
+               + f"</fib-lookup></routing></test>"
+    }
+
+    r = http_request(
+        URL,
+        'GET',
+        params=params)
+
+    if r.get("response").get("result"):
+        r = r.get("response").get("result")
+        if r.get("interface"):
+            return r
+        else:
+            raise DemistoException(f"Failed to test FIB to {dest_ip}")
+    else:
+        raise DemistoException(f"Failed to test FIB to {dest_ip}")
+
+
+def panorama_route_lookup_command():
+    """
+    Gets the outgoing interface from the Palo Alto Firewall route table
+    Must be run against a firewall instance
+    """
+    if not VSYS:
+        raise DemistoException("The 'panorama-route-lookup' command is only relevant for a Firewall instance.")
+
+    dest_ip = demisto.args().get("dest_ip")
+    vr = demisto.args().get("virtual_router", "default")
+    r = panorama_route_lookup(dest_ip, vr)
+
+    r["dest_ip"] = dest_ip
+    markdown = tableToMarkdown('Route Lookup Results', r, headers=["dest_ip", "interface"])
+    results = CommandResults(
+        readable_output=markdown,
+        outputs_prefix='Panorama.RouteLookup',
+        outputs_key_field='dest_ip',
+        outputs=r
+    )
+    return_results(results)
+    return r
+
+
+def panorama_zone_lookup_command():
+    """
+    Gets the outgoing interface from the PAN Firewall using "test routing fib-lookup", and the interfaces
+    from "show interfaces", then joins the two to determine the outgoing zone.
+    """
+    if not VSYS:
+        raise DemistoException("The 'panorama-route-lookup' command is only relevant for a Firewall instance.")
+
+    dest_ip = demisto.args().get("dest_ip")
+    vr = demisto.args().get("virtual_router", "default")
+    route = panorama_route_lookup(dest_ip, vr)
+    if not route:
+        return_results(f"Could find a matching route to {dest_ip}.")
+        return
+
+    interface = route.get("interface")
+    interfaces = panorama_get_interfaces()
+
+    r = {}
+    if "ifnet" in interfaces["response"].get("result"):
+        for entry in interfaces["response"]["result"]["ifnet"]["entry"]:
+            if entry.get("name") == interface:
+                if "zone" in entry:
+                    r = {**entry, **route}
+
+    if r:
+        # add the given dest_ip as another item int the object for simplicity
+        r["dest_ip"] = dest_ip
+        markdown = tableToMarkdown('Zone Lookup Results', r, headers=["dest_ip", "zone", "interface"])
```

Comment:
please add the rest of the headers name, ip, id, etc.

Reply:
Resolved by removing the header spec altogether and letting tableToMarkdown figure it out (there are a number of headers that may or may not appear depending on the FW config, so I figured this was the most suitable way)